lib/cdo/project_source_json.rb

```diff
@@ -0,0 +1,31 @@
+# Manipulate Source JSON for projects
+class ProjectSourceJson
+  def initialize(source_string)
+    @parsed_json = JSON.parse(source_string)
+  rescue JSON::ParserError
+    @unparseable = true
+  end
+
+  def each_animation
+    return unless animation_manifest?
+    @parsed_json['animations']['orderedKeys'].each do |key|
+      yield @parsed_json['animations']['propsByKey'][key].merge({'key' => key})
+    end
+  end
+
+  def set_animation_version(key, version)
+    @parsed_json['animations']['propsByKey'][key]['version'] = version
+  end
+
+  def to_json
+    raise Error("Can't convert unparseable body to JSON") if @unparseable
+    JSON.generate(@parsed_json)
+  end
+
+  def animation_manifest?
+    !!@parsed_json &&
+      @parsed_json['animations'] &&
+      @parsed_json['animations']['orderedKeys'] &&
+      @parsed_json['animations']['propsByKey']
+  end
+end
```

Comment:
I'm considering _also_ extracting an `AnimationManifest` model, which would make `ProjectSourceJson` a pretty thin wrapper - but might encapsulate behavior better in the future.

Reply:
I'm torn on abstracting more because ideally the way to invest further here is to move animations into the /v3/files API, but we don't really have time to pay the cost of doing that right now. The ProjectSourceJson class is a really nice improvement but I think it's fine to omit AnimationManifest for now.